src/index.tsx

```diff
@@ -142,10 +147,53 @@ const App: React.FC = () => (
   </SaleorProvider>
 );
 
+const useAuthentication = () => {
```

Comment:
**suggestion**: can you move this file into its own file?

I think `index.tsx` is a big enough right now - maybe it is worth moving this hook into its own file?